gateway/scaling/function_scaler.go

```diff
@@ -0,0 +1,108 @@
+package scaling
+
+import (
+	"fmt"
+	"log"
+	"time"
+)
+
+// NewFunctionScaler create a new scaler with the specified
+// ScalingConfig
+func NewFunctionScaler(config ScalingConfig) FunctionScaler {
```

Comment:
@LucasRoesler here it is, somehow I think I didn't delete it in the source file or hit save on it.